text_to_x/text_to_embeddings.py

```diff
@@ -0,0 +1,306 @@
+from collections import Counter
+import numpy as np
+import pandas as pd
+from scipy import sparse
+from scipy.sparse import linalg 
+import umap
+    
+    
+class SvdEmbeddings():
+    '''
+    train and operate with PMI-SVD embeddings
+    
+    Examples:
+    - training with a manually set class instance
+        your_instance = SvdEmbeddings(texts)
+        your_instance.train(front_window = int, back_window = int, embedding_dim = int)
+    
+    - training with default settings:
+        your_instance = svd2vec_run_default(texts)
+        
+    - visualizing the fit
+    '''
+    
+    def __init__(self, texts, texts_colname = 'lemma'):
+        '''
+        texts (iterable | TextToToken): input list of dataframes acquired from text_to_x.text_to_token. The "lemma" column is used. 
+        texts_colname (string): column from TextToToken DataFrame to process ("lemma" / "stem" / "word")
+        '''
+
+        self.model = {}
+        self.tok2ind = {}
+        self.ind2tok = {}
+        self.unigram_counts = {}
+        
+        # check if input is TextToToken or simialr
+        if not isinstance(texts[0], pd.DataFrame):
+            raise ValueError("Input texts are not in a DataFrame or a list of DataFrames, please use TextToToken for preprocessing")
+        
+        # unpack lemmas 
+        lemmas_to_list = []
+        for doc in texts:
+            doc_to_list = doc.loc[:, texts_colname].tolist()
+            lemmas_to_list.append(doc_to_list)
```

Comment:
Would likely be faster to do in a list comprehension. E.g.:

`
lemmas_list = [doc.loc[:, texts_colname].tolist() for doc in texts]
`